usr/lib/common/new_host.c

```diff
@@ -3380,20 +3380,17 @@ CK_RV SC_DigestEncryptUpdate(STDLL_TokData_t *tokdata,
                              CK_ULONG ulPartLen, CK_BYTE_PTR pEncryptedPart,
                              CK_ULONG_PTR pulEncryptedPartLen)
 {
-    UNUSED(sSession);
-    UNUSED(pPart);
-    UNUSED(ulPartLen);
-    UNUSED(pEncryptedPart);
-    UNUSED(pulEncryptedPartLen);
+    CK_RV rc;
 
-    if (tokdata->initialized == FALSE) {
-        TRACE_ERROR("%s\n", ock_err(ERR_CRYPTOKI_NOT_INITIALIZED));
-        return CKR_CRYPTOKI_NOT_INITIALIZED;
-    }
+    rc = SC_EncryptUpdate(tokdata, sSession, pPart, ulPartLen,
```

Comment:
I don't see the code path, where `SC_EncryptUpdate()` returns with `rc == CKR_OK` so that the second check (`pEncryptedPart == NULL`) will be true. Can you please give an example, in which situations this is the case?

Reply:
This is the case for a size-quer call. You call SC_EncryptUpdate with pEncryptedPart = NULL but pulEncryptedPartLen != NULL. This indicates that you want to know the required size of the output buffer.
The same can be used for SC_DigestEncryptUpdate. So the code must cover these size-query calls.
Note that the digest update must only happen when the encrypt step actually has produced output. 
PKCS#11:
_C_DigestEncryptUpdate uses the convention described in Section 5.2 on producing output. If a
C_DigestEncryptUpdate call does not produce encrypted output (because an error occurs, or because
pEncryptedPart has the value NULL_PTR, or because pulEncryptedPartLen is too small to hold the entire
encrypted part output), then no plaintext is passed to the active digest operation._